src/System.Linq/src/System/Linq/SelectMany.cs

```diff
@@ -231,18 +231,11 @@ public TResult[] ToArray()
                 {
                     IEnumerable<TResult> enumerable = _selector(element);
 
-                    int count;
-                    if (EnumerableHelpers.TryGetCount(enumerable, out count))
+                    // TODO: Add test to ensure the bug is fixed.
+                    if (builder.ReserveOrAdd(enumerable))
```

Comment:
Please submit a separate PR with the SelectMany bug fix.  It doesn't have anything to do with Concat.